automation/src/main/java/org/greenplum/pxf/automation/components/tinc/Tinc.java

```diff
@@ -28,8 +29,19 @@ public void init() throws Exception {
 
 		super.init();
 
-		runCommand("cd " + gphome);
-		runCommand("source greenplum_path.sh");
+		// This property is for setting PYTHONPATH in local env.
+		// This change is for running the Automation tests in GP7.
+		// Since, Tinc still runs on Python2, It won't work with Python3 libs which are present with GP7, so we need to set the PYTHONPATH
+		// pointing towards the Python2 libs.
+		String pxfAutomationTincDeps = System.getenv("PXF_AUTOMATION_TINC_DEPS");
+		String pxfAutomationTincDepsCommand;
+		if (StringUtils.isNotBlank(pxfAutomationTincDeps)) {
+			pxfAutomationTincDepsCommand = "export PYTHONPATH=" + pxfAutomationTincDeps;
+		}
+		else {
+			pxfAutomationTincDepsCommand = "export PYTHONPATH=/home/gpadmin/python";
+		}
```

Comment:
Since it looks like we still need to source `$GPHOME/greenplum_path.sh` and this will set `PYHTONPATH` to the correct location when testing against GP6, we don't need to set this variable then. Let's only override `PYTHONPATH` if the user has set it.

```suggestion
		String pxfAutomationTincDeps = System.getenv("PXF_AUTOMATION_TINC_DEPS");
		if (StringUtils.isNotBlank(pxfAutomationTincDeps)) {
			runCommand("export PYTHONPATH=" + pxfAutomationTincDeps);
		}
```